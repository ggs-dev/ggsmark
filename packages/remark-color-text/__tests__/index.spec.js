import remark from 'remark'
import html from 'remark-html'
import color from '../src'
import dedent from 'dedent'

describe('block color text with inline style', () => {
  test('should not color if token never ends', () => {
    // Arrange
    let string = dedent`
    !# red

    this should not be red

    there's no end!!!
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should short line', () => {
    // Arrange
    let string = dedent`
    !# red
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color in lowercase', () => {
    // Arrange
    let string = dedent`
    !# Red
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should not color if lowercase is turned off', () => {
    // Arrange
    let string = dedent`
    !# rEd
    123
    !#
    `

    // Act
    let result = remark()
      .use(html)
      .use(color, { lowercase: false })
      .processSync(string)
      .toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color rgba', () => {
    // Arrange
    let string = dedent`
    !# rgba(100 ,100, 100, 10)
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color rgb', () => {
    // Arrange
    let string = dedent`
    !# rgb(100 ,100, 100)
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color hex 3 characters', () => {
    // Arrange
    let string = dedent`
    !# #A0D
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color hex 6 characters', () => {
    // Arrange
    let string = dedent`
    !# #D0E10E
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should not escape', () => {
    // Arrange
    let string = dedent`
    !# #D0E10E; height: 9999px; width: 9999px;"
    123
    !#
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('multiple lines of coloring', () => {
    // Arrange
    let string = dedent`
    # Hello world before color

    before color

    !# red
    Hello I should be in red text right under the start point

    # I should still be red text

    !#

    text in between

    !# blue
    Hello I should be in blue text right under the start point

    # I should still be blue text

    more text
    I'm right above the color token and should still be blue
    !#
    I'm right under the end token
    after text

    ## Some more headings
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('inline color text with inline style', () => {
  test('should color single line', () => {
    // Arrange
    let string = dedent`
    !# red (this is inline!)
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should not color if not closed', () => {
    // Arrange
    let string = dedent`
    !# red (this is inline!
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should color between text', () => {
    // Arrange
    let string = dedent`
    Test

    Lorem ipsum dolor sit !# red (**Vestibulum cursus volutpat auctor**), consectetur adipiscing elit.
    Vestibulum cursus volutpat auctor. !# #D0E10E (Pellentesque euismod) ipsum placerat arcu condimentum mattis.

    Lorem ipsum dolor sit !# red (amet), consectetur adipiscing elit.
    Vestibulum cursus volutpat auctor. !# #D0E10E (Pellentesque euismod) ipsum placerat arcu condimentum mattis.
    `

    // Act
    let result = remark().use(html).use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should not split by nbsp character', () => {
    // Arrange
    let string = `!# #D0E10E (hello&nbsp;world)`

    // Act
    let result = remark().use(color).processSync(string).toString()

    // Assert
    expect(result).toMatchSnapshot()
  })
})
