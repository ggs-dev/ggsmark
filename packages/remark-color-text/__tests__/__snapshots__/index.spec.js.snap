// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block color text with inline style multiple lines of coloring 1`] = `
"<h1>Hello world before color</h1>
<p>before color</p>
<div style=\\"color: red\\"><p>Hello I should be in red text right under the start point</p><h1>I should still be red text</h1></div>
<p>text in between</p>
<div style=\\"color: blue\\"><p>Hello I should be in blue text right under the start point</p><h1>I should still be blue text</h1><p>more text
I'm right above the color token and should still be blue</p></div>
<p>I'm right under the end token
after text</p>
<h2>Some more headings</h2>
"
`;

exports[`block color text with inline style should color hex 3 characters 1`] = `
"<div style=\\"color: #a0d\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should color hex 6 characters 1`] = `
"<div style=\\"color: #d0e10e\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should color in lowercase 1`] = `
"<div style=\\"color: red\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should color rgb 1`] = `
"<div style=\\"color: rgb(100 ,100, 100)\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should color rgba 1`] = `
"<div style=\\"color: rgba(100 ,100, 100, 10)\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should not color if lowercase is turned off 1`] = `
"<div style=\\"color: rEd\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should not color if token never ends 1`] = `
"<p>!# red</p>
<p>this should not be red</p>
<p>there's no end!!!</p>
"
`;

exports[`block color text with inline style should not escape 1`] = `
"<div style=\\"color: #d0e10e\\"><p>123</p></div>
"
`;

exports[`block color text with inline style should short line 1`] = `
"<div style=\\"color: red\\"><p>123</p></div>
"
`;

exports[`inline color text with inline style should color between text 1`] = `
"<p>Test</p>
<p>Lorem ipsum dolor sit <span style=\\"color: red\\"><strong>Vestibulum cursus volutpat auctor</strong></span>, consectetur adipiscing elit.
Vestibulum cursus volutpat auctor. <span style=\\"color: #d0e10e\\">Pellentesque euismod</span> ipsum placerat arcu condimentum mattis.</p>
<p>Lorem ipsum dolor sit <span style=\\"color: red\\">amet</span>, consectetur adipiscing elit.
Vestibulum cursus volutpat auctor. <span style=\\"color: #d0e10e\\">Pellentesque euismod</span> ipsum placerat arcu condimentum mattis.</p>
"
`;

exports[`inline color text with inline style should color single line 1`] = `
"<p><span style=\\"color: red\\">this is inline!</span></p>
"
`;

exports[`inline color text with inline style should not color if not closed 1`] = `
"<p>!# red (this is inline!</p>
"
`;

exports[`inline color text with inline style should not split by nbsp character 1`] = `
"!# #d0e10e (helloÂ world)
"
`;
